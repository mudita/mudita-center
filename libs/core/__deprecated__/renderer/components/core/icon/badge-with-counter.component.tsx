/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import * as React from "react"
import styled from "styled-components"
import { FunctionComponent } from "Core/core/types/function-component.interface"
import { getTextStyles, TextDisplayStyle } from "Core/__deprecated__/renderer/components/core/text/text.component"

interface Props {
  indicator: number
}

const BadgeWithCounterContainer = styled.span`
  ${getTextStyles(TextDisplayStyle.Headline5)};
  width: 2rem;
  height: 2rem;
  background-color: black;
  color: white;
  border-radius: 50%;
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  right: 0;
  transform: translate(calc(50% + -0.2rem), calc(-50% + 0.8rem));
`
export const BadgeWithCounter: FunctionComponent<Props> = ({ indicator }) => {
  return <BadgeWithCounterContainer>{indicator}</BadgeWithCounterContainer>
}
