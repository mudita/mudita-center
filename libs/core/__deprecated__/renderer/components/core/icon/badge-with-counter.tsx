/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import * as React from "react"
import styled from "styled-components"
import { fontWeight } from "Core/core/styles/theming/theme-getters"
import { FunctionComponent } from "Core/core/types/function-component.interface"

interface Props {
  indicator: number
}

const BadgeWithCounterContainer = styled.span`
  width: 2rem;
  height: 2rem;
  background-color: black;
  color: white;
  border-radius: 50%;
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  right: 0;
  font-size: 1.2rem;
  font-weight: ${fontWeight("bold")};
  transform: translate(calc(50% + -0.2rem), calc(-50% + 0.8rem));
`
export const BadgeWithCounter: FunctionComponent<Props> = ({ indicator }) => {
  return <BadgeWithCounterContainer>{indicator}</BadgeWithCounterContainer>
}
