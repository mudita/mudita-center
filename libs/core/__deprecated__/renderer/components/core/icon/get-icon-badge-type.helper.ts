/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { IconBadgeType } from "Core/__deprecated__/renderer/components/core/icon/icon-badge-type.constant"

export const getIconBadgeType = (
  badge: undefined | boolean | IconBadgeType,
): IconBadgeType | undefined => {
  if (badge === true || badge === IconBadgeType.Badge) {
    return IconBadgeType.Badge
  } else if (badge === IconBadgeType.BadgeWithCounter) {
    return IconBadgeType.BadgeWithCounter
  } else {
    return undefined
  }
}
