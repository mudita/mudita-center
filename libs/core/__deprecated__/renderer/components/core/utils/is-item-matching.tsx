/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

// AUTO DISABLED - fix me if you like :)
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type IsItemMatching = (item: any, searchString: string) => boolean
export const isItemValueMatching: IsItemMatching = (
  itemValue: string,
  search
) => itemValue.toLowerCase().includes(search.toLowerCase())
