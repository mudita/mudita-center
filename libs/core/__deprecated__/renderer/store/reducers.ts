/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { combineReducers } from "redux"
import { deviceReducer } from "Core/device/reducers"
import { crashDumpReducer } from "Core/crash-dump/reducers"
import { backupReducer } from "Core/backup/reducers"
import { messagesReducer } from "Core/messages/reducers"
import { modalsManagerReducer } from "Core/modals-manager/reducers"
import { contactSupportReducer } from "Core/contact-support/reducers"
import { contactsReducer } from "Core/contacts/reducers"
import { dataSyncReducer } from "Core/data-sync/reducers"
import { filesManagerReducer } from "Core/files-manager/reducers"
import { notificationReducer } from "Core/notification/reducers"
import { templateReducer } from "Core/templates/reducers"
import { newsReducer } from "Core/news/reducers"
import { settingsReducer } from "Core/settings/reducers"
import { updateOsReducer } from "Core/update/reducers"
import { discoveryDeviceReducer } from "Core/discovery-device/reducers/discovery-device.reducer"
import { deviceInitializationReducer } from "Core/device-initialization/reducers/device-initialization.reducer"
import { appInitializationReducer } from "Core/app-initialization/reducers/app-initialization.reducer"
import { coreDeviceReducer } from "core-device/feature"
import { deviceManagerReducer } from "device-manager/feature"
import {
  dataMigrationReducer,
  genericBackupsReducer,
  genericDataTransferReducer,
  genericFileTransferReducer,
  genericModalsReducer,
  genericViewsReducer,
  importsReducer,
  externalProvidersReducer
} from "generic-view/store"
import { appStateReducer } from "shared/app-state"

export const reducers = {
  device: deviceReducer,
  backup: backupReducer,
  crashDump: crashDumpReducer,
  messages: messagesReducer,
  contacts: contactsReducer,
  dataSync: dataSyncReducer,
  modalsManager: modalsManagerReducer,
  contactSupport: contactSupportReducer,
  filesManager: filesManagerReducer,
  notification: notificationReducer,
  templates: templateReducer,
  news: newsReducer,
  settings: settingsReducer,
  update: updateOsReducer,
  discoveryDevice: discoveryDeviceReducer,
  deviceInitialization: deviceInitializationReducer,
  appInitialization: appInitializationReducer,
  coreDevice: coreDeviceReducer,
  deviceManager: deviceManagerReducer,
  genericViews: genericViewsReducer,
  genericModals: genericModalsReducer,
  genericBackups: genericBackupsReducer,
  genericFileTransfer: genericFileTransferReducer,
  genericImport: importsReducer,
  externalProviders: externalProvidersReducer,
  appState: appStateReducer,
  dataMigration: dataMigrationReducer,
  genericDataTransfer: genericDataTransferReducer,
}

export const combinedReducers = combineReducers(reducers)
