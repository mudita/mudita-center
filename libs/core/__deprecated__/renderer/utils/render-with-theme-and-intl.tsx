/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import ReactModal from "react-modal"
import { ThemeProvider } from "styled-components"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Core/__deprecated__/renderer/locales/default/en-US.json"
import extractLanguageKeys from "Core/__deprecated__/renderer/utils/extract-test-locale"
import theme from "Core/core/styles/theming/theme"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

// AUTO DISABLED - fix me if you like :)
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function constructWrapper(ui: React.ReactElement) {
  return (
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>
  )
}

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    constructWrapper(ui),
    // AUTO DISABLED - fix me if you like :)
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    options
  )
}
