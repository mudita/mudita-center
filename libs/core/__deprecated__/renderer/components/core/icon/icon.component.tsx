/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import * as React from "react"
import {
  getEnumName,
  getIconType,
} from "Core/__deprecated__/renderer/components/core/icon/icon.config"
import Svg from "Core/__deprecated__/renderer/components/core/svg/svg.component"
import { backgroundColor } from "Core/core/styles/theming/theme-getters"
import { FunctionComponent } from "Core/core/types/function-component.interface"
import styled, { css } from "styled-components"
import { IconType } from "Core/__deprecated__/renderer/components/core/icon/icon-type"
import { BadgeWithCounter } from "Core/__deprecated__/renderer/components/core/icon/badge-with-counter.component"
import { getIconBadgeType } from "Core/__deprecated__/renderer/components/core/icon/get-icon-badge-type.helper"
import { IconBadgeType } from "Core/__deprecated__/renderer/components/core/icon/icon-badge-type.constant"

export enum IconSize {
  Small = 1.6,
  Medium = 2.4,
  Big = 2.8,
  Large = 3.2,
  ExtraLarge = 3.6,
  Enormous = 4.8,
}

export interface Props {
  badge?: boolean | IconBadgeType
  badgeCountIndicator?: number
  height?: number
  width?: number
  size?: IconSize
  type?: IconType
  disabled?: boolean
}

const badgeStyles = css`
  &:after {
    display: block;
    content: "";
    position: absolute;
    top: -0.2rem;
    right: -0.3rem;
    height: 0.8rem;
    width: 0.8rem;
    border-radius: 50%;
    background-color: ${backgroundColor("activity")};
  }
`

const Wrapper = styled.span<{
  badge: boolean
  height?: number
  width?: number
  size?: IconSize
}>`
  display: flex;
  justify-content: center;
  align-items: center;
  height: ${({ height = 2, size }) => (size ? `${size}rem` : `${height}rem`)};
  width: ${({ width = 2, size }) => (size ? `${size}rem` : `${width}rem`)};
  position: relative;
  font-style: normal;

  svg {
    width: 100%;
    height: 100%;
  }

  ${({ badge }) => badge && badgeStyles};
`

const Icon: FunctionComponent<Props> = ({
  badge = false,
  badgeCountIndicator = 0,
  className,
  size,
  height,
  width,
  type,
  disabled,
  ...rest
}) => {
  const iconBadgeType = getIconBadgeType(badge)
  return (
    <Wrapper
      badge={iconBadgeType === IconBadgeType.Badge}
      data-testid={`icon-${getEnumName(type) ?? ""}`}
      className={className}
      height={size || height || width}
      width={size || width || height}
      {...rest}
    >
      {iconBadgeType === IconBadgeType.BadgeWithCounter && (
        <BadgeWithCounter indicator={badgeCountIndicator} disabled={disabled} />
      )}
      <Svg Image={getIconType(type)} />
    </Wrapper>
  )
}

export default Icon
