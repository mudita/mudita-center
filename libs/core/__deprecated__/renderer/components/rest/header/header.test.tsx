/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import React from "react"
import { MemoryRouter } from "react-router-dom"
import Header from "Core/__deprecated__/renderer/components/rest/header/header.component"
import Tabs from "Core/__deprecated__/renderer/components/rest/header/tabs.component"
import { renderWithThemeAndIntl } from "Core/__deprecated__/renderer/utils/render-with-theme-and-intl"
import { HeaderButton } from "Core/__deprecated__/renderer/wrappers/layout-desktop-wrapper"
import { intl } from "Core/__deprecated__/renderer/utils/intl"
import { flags } from "Core/feature-flags"
import { IconType } from "Core/__deprecated__/renderer/components/core/icon/icon-type"
import { Provider } from "react-redux"
import store from "Core/__deprecated__/renderer/store"

jest.mock("Core/feature-flags")

test("matches snapshot without tabs", () => {
  const currentLocation = "/overview"
  const { container } = renderWithThemeAndIntl(
    <Provider store={store}>
      <MemoryRouter initialEntries={[currentLocation]}>
        <Header middleComponent={<Tabs currentLocation={currentLocation} />} />
      </MemoryRouter>
    </Provider>
  )
  const header = container.firstChild
  expect(header).toMatchSnapshot()
})

test("matches snapshot with tabs", () => {
  jest.spyOn(flags, "get").mockReturnValueOnce(true)
  const currentLocation = "/phone"
  const { container } = renderWithThemeAndIntl(
    <Provider store={store}>
      <MemoryRouter initialEntries={[currentLocation]}>
        <Header middleComponent={<Tabs currentLocation={currentLocation} />} />
      </MemoryRouter>
    </Provider>
  )
  const header = container.firstChild
  expect(header).toMatchSnapshot()
})

test("header should render correct pathname", () => {
  const { getByTestId } = renderWithThemeAndIntl(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/overview"]}>
        <Header />
      </MemoryRouter>
    </Provider>
  )
  expect(getByTestId("location")).toHaveTextContent("module.overview")
})

test("button renders on news page", () => {
  const currentLocation = "/news"
  const buttonsIconId = "icon-ExternalLink"
  const { getByTestId } = renderWithThemeAndIntl(
    <Provider store={store}>
      <MemoryRouter initialEntries={[currentLocation]}>
        <Header
          middleComponent={<Tabs currentLocation={currentLocation} />}
          button={
            <HeaderButton
              Icon={IconType.ExternalLink}
              label={intl.formatMessage({
                id: "module.news.moreNewsButtonLabel",
              })}
              href={"https://www.mudita.com/"}
              target="_blank"
            />
          }
        />
      </MemoryRouter>
    </Provider>
  )
  expect(getByTestId(buttonsIconId)).toBeInTheDocument()
})

test("button doesnt render on other pages than news", () => {
  const currentLocation = "/phone"
  const buttonsIconId = "icon-More"
  const { queryByTestId } = renderWithThemeAndIntl(
    <Provider store={store}>
      <MemoryRouter initialEntries={[currentLocation]}>
        <Header middleComponent={<Tabs currentLocation={currentLocation} />} />
      </MemoryRouter>
    </Provider>
  )
  expect(queryByTestId(buttonsIconId)).not.toBeInTheDocument()
})
