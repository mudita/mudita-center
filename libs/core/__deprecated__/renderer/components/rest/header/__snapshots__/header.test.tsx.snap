// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot without tabs 1`] = `
.c1 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.5;
  font-weight: 700;
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  color: #000000;
  white-space: pre-wrap;
}

.c0 {
  display: grid;
  height: 100%;
  grid-template-columns: 1fr 1fr 1fr;
  background-color: #ffffff;
  border-bottom: 0.1rem solid #d2d6db;
}

.c2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-left: 3.2rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <h4
    class="c1 c2"
    color="primary"
    data-testid="location"
    id="app-header"
  >
    [value] module.overview
  </h4>
  <div
    class="c3"
  />
</div>
`;
